js/wp-seo-shortcode-plugin.js

```diff
@@ -0,0 +1,281 @@
+	this.checkShortcodes( meta );
+
+	this.parseShortcodes( this.declareReady );
+};
+
+/**
+ * Bind elements to be able to reload the dataset if shortcodes get added.
+ */
+YoastShortcodePlugin.prototype.bindElementEvents = function() {
+	var elems = [ 'title', 'snippet_title', 'snippet_meta', 'content' ];
+
+	for ( var i = 0; i < elems.length; i++ ) {
+		document.getElementById( elems[ i ] ).addEventListener( 'keydown', yoastShortcodePlugin.inputEventCallback );
+		document.getElementById( elems[ i ] ).addEventListener( 'input', yoastShortcodePlugin.inputEventCallback );
+	}
+};
+
+/**
+ * Callback function for events on input elements. Beware! This function is always called within the scope of the element!
+ *
+ * @param {object} ev The event object
+ */
+YoastShortcodePlugin.prototype.inputEventCallback = function( ev ) {
+	var text;
+
+	if ( ev.currentTarget.id === 'content' ) {
+		text = yoastShortcodePlugin.getContentTinyMCE();
+	}
+	else if ( ev.currentTarget.tagName === 'SPAN' ) {
+		text = ev.currentTarget.textContent;
+	}
+	else {
+		text = ev.currentTarget.value;
+	}
+
+	yoastShortcodePlugin.checkShortcodes( text );
+	yoastShortcodePlugin.parseShortcodes( yoastShortcodePlugin.declareReloaded );
+};
+
+/**
+ * gets content from the content field, if tinyMCE is initialized, use the getContent function to get the data from tinyMCE
+ * @returns {String}
+ */
+YoastShortcodePlugin.prototype.getContentTinyMCE = function() {
+	var val = document.getElementById( 'content' ).value;
+	if ( tinyMCE.editors.length !== 0 ) {
+		val = tinyMCE.get( 'content' ).getContent();
+	}
+
+	return val;
+};
+
+/* SHORTCODE PARSING */
+
+/**
+ * Add a shortcode to the unparsed shortcodes unless was already parsed.
+ *
+ * @param {string} shortcode
+ */
+YoastShortcodePlugin.prototype.checkShortcode = function( shortcode ) {
+	var already_exists = false;
+
+	for ( var i in this.parsedShortcodes ) {
+		if (this.parsedShortcodes[ i ].shortcode === shortcode) {
+			already_exists = true;
+		}
+	}
+
+	if ( already_exists === false ) {
+		this.unparsedShortcodes.push( shortcode );
+	}
+};
+
+/**
+ * Checks a piece of text for shortcodes and adds them to the unparsed shortcodes if they haven't been parsed yet.
+ *
+ * @param {string} text
+ * @returns {boolean}
+ */
+YoastShortcodePlugin.prototype.checkShortcodes = function( text ) {
+	if ( typeof text !== 'string') {
+		console.error( 'Failed to get unparsed shortcodes. Expected parameter to be a string, instead received' + typeof text );
+		return false;
+	}
+
+	var shortcodes = this.matchShortcodes( text );
+
+	for ( var i in shortcodes ) {
+		var shortcode = shortcodes[ i ];
+		if ( this.unparsedShortcodes.indexOf( shortcode ) === -1 ) {
+			this.checkShortcode( shortcode );
+		}
+	}
+
+	return true;
+};
+
+/**
+ * Matches a text for shortcodes.
+ *
+ * @param {string} text
+ * @returns {array} The matched shortcodes
+ */
+YoastShortcodePlugin.prototype.matchShortcodes = function( text ) {
```

Comment:
This would be more clear if it were split into a `matchCapturingShortcodes` function and a `matchNonCapturingShortcodes` function.

So we would rename `getCapturesFromText` to `matchCapturingShortcodes` and make a new `matchNonCapturingShortcodes`.